buildSrc/subprojects/kotlin-dsl/src/main/kotlin/parser/KotlinSourceParser.kt

```diff
@@ -48,30 +67,49 @@ class KotlinSourceParser {
             ktFiles.map(block)
         }
 
+    fun parseSourceRoots(sourceRoots: List<File>, compilationClasspath: List<File>): ParsedKotlinFiles =
+        Disposer.newDisposable().let { disposable ->
+            ParsedKotlinFiles(disposable.parseKotlinFiles(sourceRoots, compilationClasspath), disposable)
+        }
+
     private
-    fun <T : Any> withParsedKotlinSource(sourceRoots: List<File>, block: (List<KtFile>) -> T) = withRootDisposable {
+    fun <T : Any?> withParsedKotlinSource(sourceRoots: List<File>, block: (List<KtFile>) -> T) =
+        Disposer.newDisposable().use {
+            parseKotlinFiles(sourceRoots, emptyList()).let(block)
+        }
 
+    private
+    fun Disposable.parseKotlinFiles(sourceRoots: List<File>, compilationClasspath: List<File>): List<KtFile> {
         val configuration = CompilerConfiguration().apply {
 
             put(CLIConfigurationKeys.MESSAGE_COLLECTOR_KEY, messageCollector)
             put(JVMConfigurationKeys.RETAIN_OUTPUT_IN_MEMORY, false)
+            put(JVMConfigurationKeys.DISABLE_OPTIMIZATION, true)
             put(CommonConfigurationKeys.MODULE_NAME, "parser")
 
+            // This is the same as in the `kotlin-library` plugin in this build
```

Comment:
⭕️ Maybe add a note in `kotlin-library` that when it changes, this should change as well?

Reply:
I extracted both to a single internal place instead, they are in the same module.

Reply:
Even better!